android/src/main/java/com/bitmovin/player/reactnative/RNPlayerView.kt

```diff
@@ -283,10 +284,10 @@ class RNPlayerView(
      * @param event Optional js object to be sent as payload.
      */
     private inline fun <reified E : Event> emitEvent(name: String, event: E) {
-        val payload = if (event is PlayerEvent) {
-            JsonConverter.fromPlayerEvent(event)
-        } else {
-            JsonConverter.fromSourceEvent(event as SourceEvent)
+        val payload = when (event) {
+            is PlayerEvent -> event.fromPlayerEvent()
+            is SourceEvent -> event.fromSourceEvent()
+            else -> throw IllegalArgumentException()
```

Comment:
Is this throw safe ? Can we have a complete when instead? (is Event a sealed class?)

Reply:
It looks like exhaustive `when` is only possible in the same module as the base class? :thinking: Or at least I did not manage to make it compile with `is OflineEvent`.